packages/pigeon/tool/shared/test_suites.dart

```diff
@@ -313,9 +314,23 @@ Future<int> _runWindowsUnitTests() async {
     return compileCode;
   }
 
-  return runProcess(
-      '$examplePath/build/windows/plugins/test_plugin/Debug/test_plugin_test.exe',
-      <String>[]);
+  // Depending on the Flutter version, the build output path is different.
+  // Check for the new location first, and fall back to the old location, to
+  // support running against both stable and master.
+  // TODO(stuartmorgan): Remove all this when these tests no longer need to
+  // support Flutter 3.13, and just construct the version of the path with the
+  // architecture.
+  const String buildDirBase = '$examplePath/build/windows';
+  const String buildRelativeBinaryPath =
+      'plugins/test_plugin/Debug/test_plugin_test.exe';
+  final String arch = Abi.current() == Abi.windowsArm64 ? 'arm64' : 'x64';
```

Comment:
FYI, the tool does not support targeting Windows Arm64 yet. Currently, building on Windows Arm64 will create an x64 executable that will run using emulation. Here's the issue that tracks creating Windows Arm64 executables: https://github.com/flutter/flutter/issues/129807

Reply:
Thanks, I misremembered what the roadmap ordering was on the arm64 support.

To avoid having a mini fire-drill when https://github.com/flutter/flutter/issues/129807 lands, I've made the scripts messier for now but future-proof, so that they will work with the new structure with only x64, but then use the new structure with arm64 when it starts working.